src/cr/cube/cube.py

```diff
@@ -410,16 +414,23 @@ def _ca_as_0th(self):
     @lazyproperty
     def _cube_dict(self):
         """dict containing raw cube response, parsed from JSON payload."""
+        cube_dict = copy.deepcopy(self._cube_response)
+        if self._mean_subvariables:
+            dimensions = cube_dict.get("result", {}).get("dimensions", {})
```

Comment:
This is interesting, but we _have_ to have both the `"result"` and the `"dimensions"` in it. And if we don't, then for the result it's ok to use `{}`, but for the dimensions it has to be `[]`, because you're appending in the next line. So, either do that, or access the values directly with square brackets (and if you want to do a check if everything is valid, do so at construction time (fail sooner rather than later)).

Reply:
u right